### PR TITLE
[SYCL] Fix OpenCL version check when verifying SPIR-V online compilation support

### PR DESCRIPTION
**Pull request by dnmokhov**
Enumerate through historic OpenCL versions, like in the plugin.

### CODE REVIEW

File: sycl/source/detail/program_manager/program_manager.cpp [hunk 1]
```diff
@@ -220,17 +220,17 @@ getOrBuild(KernelProgramCache &KPCache, KeyT &&CacheKey, AcquireFT &&Acquire,
 
 static bool isDeviceBinaryTypeSupported(const context &C,
                                         RT::PiDeviceBinaryType Format) {
+  // All formats except PI_DEVICE_BINARY_TYPE_SPIRV are supported.
```
Comment (kbobrovs) [inline sycl/source/detail/program_manager/program_manager.cpp:223]: I think that decision whether a binary type is supported by a particular context or not should be made by plugin (or BE) decision associated with the context. Looks like we need a new PI function. Can you please add a TODO?

File: sycl/source/detail/program_manager/program_manager.cpp [hunk 2]
```diff
@@ -240,9 +240,14 @@ static bool isDeviceBinaryTypeSupported(const context &C,
...
   // OpenCL 2.1 and greater require clCreateProgramWithIL
-  if ((ContextBackend == backend::opencl) &&
-      C.get_platform().get_info<info::platform::version>() >= "2.1")
-    return true;
+  if (ContextBackend == backend::opencl) {
+    // According to the spec, the version string has the following format:
+    // OpenCL<space><major_version.minor_version><space><platform-specific info>
+    auto VersionStringVector =
+        split_string(C.get_platform().get_info<info::platform::version>(), ' ');
+    if ((VersionStringVector.size() > 1) && (VersionStringVector[1] >= "2.1"))
```
Comment (kbobrovs) [inline sycl/source/detail/program_manager/program_manager.cpp:248]: Lexicographical comparison is unreliable here. I guess "12.1" >= "2.1" would evaluate to false.

**Review by kbobrovs — CHANGES_REQUESTED** [review]

**Review by s-kanaev — APPROVED** [review]

**Comment by s-kanaev** [comment]
@kbobrovs  ping

**Review by kbobrovs — APPROVED** [review]
I guess there are no versions like OpenCL 1.10, so this should work.

**Pull request merged**